From aed9402d6282c193ac23252a8686394ac833b713 Mon Sep 17 00:00:00 2001
From: Duncan Overbruck <mail@duncano.de>
Date: Wed, 4 Mar 2020 23:39:27 +0100
Subject: [PATCH 1/2] lib/transaction_check_revdeps.c: fix regression in
 revdeps check

this is probably the same as reported in #245.

Its again not resetting/initializing a variable in a loop,
falling back to the previous set value.
---
 lib/transaction_check_revdeps.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git lib/transaction_check_revdeps.c lib/transaction_check_revdeps.c
index 211f1df0..7e30e3e8 100644
--- lib/transaction_check_revdeps.c
+++ lib/transaction_check_revdeps.c
@@ -172,6 +172,8 @@ xbps_transaction_check_revdeps(struct xbps_handle *xhp, xbps_array_t pkgs)
 				error = true;
 				goto out;
 			}
+
+			ttype = 0;
 			if ((revpkgd = xbps_find_pkg_in_array(pkgs, pkgname, 0))) {
 				ttype = xbps_transaction_pkg_type(revpkgd);
 				if (ttype == XBPS_TRANS_REMOVE)
-- 
2.25.1

